pom.xml

```diff
@@ -465,51 +486,109 @@
+                    <version>1.5.3</version>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-scm-publish-plugin</artifactId>
+                    <version>1.1</version>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-toolchains-plugin</artifactId>
+                    <version>1.1</version>
+                </plugin>
+                <plugin>
+                    <groupId>com.github.ekryd.sortpom</groupId>
+                    <artifactId>sortpom-maven-plugin</artifactId>
+                    <version>${dep.plugin.sortpom.version}</version>
+                    <configuration>
+                        <createBackupFile>false</createBackupFile>
+                        <expandEmptyElements>false</expandEmptyElements>
+                        <keepBlankLines>true</keepBlankLines>
+                        <lineSeparator>\n</lineSeparator>
+                        <nrOfIndentSpace>4</nrOfIndentSpace>
+                        <sortDependencies>scope</sortDependencies>
+                        <sortProperties>true</sortProperties>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-pmd-plugin</artifactId>
+                    <configuration>
+                        <linkXRef>false</linkXRef>
+                        <analysisCache>true</analysisCache>
+                        <verbose>true</verbose>
+                        <rulesets>
+                            <ruleset>${maven.multiModuleProjectDirectory}/src/build/pmd.xml</ruleset>
+                        </rulesets>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-checkstyle-plugin</artifactId>
+                    <configuration>
+                        <configLocation>${maven.multiModuleProjectDirectory}/src/build/checkstyle.xml</configLocation>
+                        <suppressionsLocation>${maven.multiModuleProjectDirectory}/src/build/checkstyle-suppress.xml</suppressionsLocation>
+                        <sourceDirectories>${project.build.sourceDirectory},${project.build.testSourceDirectory}</sourceDirectories>
+                    </configuration>
+                </plugin>
             </plugins>
         </pluginManagement>
         <plugins>
             <plugin>
                 <groupId>com.github.ekryd.sortpom</groupId>
                 <artifactId>sortpom-maven-plugin</artifactId>
-                <version>2.8.0</version>
                 <executions>
                     <execution>
-                        <phase>verify</phase>
+                        <phase>compile</phase>
```

Comment:
I'm curious why we're moving sortpom to the compile phase, and moving kotlin plugin from compile to process-sources phase?

Reply:
sortpom-compile seemed like a logical fit to me. Compiling is when creation/modification of files happens... Having files changed by running tests or in the verify phase seems more of a mismatch. I can link it to any other phase if you disagree though. I guess I also specified a phase explicitly because I'm not a big fan of just using implicit defaults and not actually knowing/seeing the real config...

The kotlin thing was a jetbrains recommendation, like the commit message says:
![image](https://user-images.githubusercontent.com/3950300/40995606-c89e8610-68ff-11e8-9331-96ef70a65599.png)

I guess in this case it makes no practical difference, the compilation already worked, I guess there are no references to any kotlin files from the java files... But I like to pre-emptively follow general recommendations. Again though, you're the boss of course.